python/cudf/cudf/core/column/numerical.py

```diff
@@ -641,6 +641,17 @@ def to_pandas(
             pd_series.index = index
         return pd_series
 
+    def _reduction_result_dtype(self, reduction_op: str) -> Dtype:
+        col_dtype = self.dtype
+        if reduction_op in {"sum", "product"}:
+            col_dtype = (
+                col_dtype if col_dtype.kind == "f" else np.dtype("int64")
```

Comment:
So pandas won't upcast float32 results to float64 to prevent overflow?

Reply:
That is my observation, yes.

```python
>>> type(pd.Series([1,2,3], dtype='int8').sum())
<class 'numpy.int64'>
>>> type(pd.Series([1.0, 2.0, 3.0], dtype='float32').sum())
<class 'numpy.float32'>
>>> type(pd.Series([1.0, 2.0, 3.0], dtype='float64').sum())
<class 'numpy.float64'>
```